x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelDefinition.java

```diff
@@ -169,8 +188,71 @@ public TrainedModelDefinition build() {
             if (preProcessors != null && preProcessors.size() > 1 && processorsInOrder == false) {
                 throw new IllegalArgumentException("preprocessors must be an array of preprocessor objects");
             }
-            return new TrainedModelDefinition(this.trainedModel, this.preProcessors);
+            return new TrainedModelDefinition(this.trainedModel, this.preProcessors, this.input);
         }
     }
 
+    public static class Input implements ToXContentObject, Writeable {
+
+        public static final String NAME = "trained_mode_definition_input";
+        public static final ParseField FIELD_NAMES = new ParseField("field_names");
```

Comment:
Is it possible to make server-side and client-side names consistent? I.e. `feature_names` vs `field_names`?